source/opt/passes.h

```diff
@@ -74,6 +75,15 @@ class EliminateDeadConstantPass : public Pass {
   bool Process(ir::Module*) override;
 };
 
+// The optimization pass to deduplicate the constants. Front-end constants with
+// same value will be unified and only one constants will be kept for each
```

Comment:
Float point values with different bit patterns are considered different.

Updated in 5d7fa1e
Now, constants with any decorations on their result ids or type ids won't be processed any more. They won't be used to replace other constants, and other constants won't replace them neither (even though the decorations may be the same).